signac_dashboard/dashboard.py

```diff
@@ -473,17 +496,22 @@ def prevent_caching(response):
 
         @dashboard.app.context_processor
         def injections():
-            session.setdefault(
-                "enabled_modules",
-                [i for i in range(len(self.modules)) if self.modules[i].enabled],
-            )
+            keyfunc = lambda m: m.context
+            grouped_modules = groupby(sorted(self.modules, key=keyfunc), key=keyfunc)
```

Comment:
This is pre-calculated as `self.modules_by_context`, right? `self.modules` shouldn't be allowed to change during dashboard execution iirc. If so, we don't need to do the sort/groupby again. Just re-use the existing dict.

Reply:
Yep! I didn't realize how `injections` worked so I was adding things in both places while learning.